stream_alert/rule_processor/rules_engine.py

```diff
@@ -235,7 +235,7 @@ def process_subkeys(record, payload_type, rule):
                     'The required subkey %s is not found when trying to process %s: \n%s',
                     key, rule.name, record)
                 return False
-            if not all(record[key].get(x) for x in nested_keys):
+            if any(x not in record[key] for x in nested_keys):
```

Comment:
I meant to respond on the issue you filed, but this is also be my preferred fix to this 👍

Reply:
I'm also just realizing that we don't do anything to check that `record[key]` is a dict 🤔 I suppose it should be up to the rule-writer to know that keys in their log will reference dicts (objects) and not use, say, a top level key that references some other type. that said, it might not hurt to add that safety check here..